### DISCUSSION TITLE
Switching between authed users throws an error and causes a full load in SvelteKit

### DISCUSSION THREAD

**Discussion by babakfp** [pocketbase/pocketbase#4231] [Q&A] [answered]
Hi

Am I doing something wrong on the Pocketbase side?

https://github.com/pocketbase/pocketbase/assets/44144724/c6136e81-ffaa-4f00-968d-1e380fe23db5

![image](https://github.com/pocketbase/pocketbase/assets/44144724/7cdf37f8-151a-4819-bc59-2f6790364baa)

`src/routes/(other)/login-as/+server.ts`:

```ts
import { redirect } from "@sveltejs/kit"

const auth = {
    user: {
        username: "user",
        password: "useruser",
    },
    admin: {
        username: "admin",
        password: "adminadmin",
    },
}

export const GET = async ({ locals }) => {
    if (!locals.user) redirect(303, "/login")

    if (locals.user.isAdmin) {
        await locals.pb
            .collection("users")
            .authWithPassword(auth.user.username, auth.user.password)
    } else {
        await locals.pb
            .collection("users")
            .authWithPassword(auth.admin.username, auth.admin.password)
    }

    redirect(303, "/")
}
```

**Comment by ganigeorgiev** [comment]
There is not enough details to understand what is the issue or what is causing it.

How is `locals.pb` initialized? Are those user credentials always fixed and global?

You can try to wrap the `if` section with a `try/catch` and inspect the error with `console.log(err.response, err.originalError)` (see also [Error handling](https://github.com/pocketbase/js-sdk?tab=readme-ov-file#error-handling)).

If the request is rejected by the PocketBase server, you can try to inspect the error log in the _Admin UI > Logs_ and/or start the executable with `--dev`.

**Reply by babakfp** [reply]
PocketBase itself doesn't throw any errors, neither in the browser nor the server console, and neither in the logs. Also, that error and the warning (that I sent a screenshot) only happen on the client side and nothing on the server console. I guess this is not related to PocketBase then. Thanks.

**Comment by babakfp** [comment, marked as answer]
I switched to using form actions and it fixed the issue. I'm not sure if I changed anything else too, but whenever I switch to the above approach, it logs the same warning and error.
